fdbserver/masterserver.actor.cpp

```diff
@@ -1228,6 +1228,26 @@ ACTOR Future<Void> provideVersions(Reference<MasterData> self) {
 	}
 }
 
+void updateLiveCommittedVersion(Reference<MasterData> self, ReportRawCommittedVersionRequest req) {
+	self->minKnownCommittedVersion = std::max(self->minKnownCommittedVersion, req.minKnownCommittedVersion);
+	if (req.version > self->liveCommittedVersion.get()) {
+		// Note this set call switches context to any waiters on liveCommittedVersion before continuing.
+		self->liveCommittedVersion.set(req.version);
+		if (self->liveCommittedVersion.get() <= req.version) {
+			self->databaseLocked = req.locked;
+			self->proxyMetadataVersion = req.metadataVersion;
+		}
```

Comment:
As I think more about this, an easy solution is to just move `set` to be the last thing we do here, e.g.,
```
		self->databaseLocked = req.locked;
		self->proxyMetadataVersion = req.metadataVersion;
		self->liveCommittedVersion.set(req.version);
```
In this way, we won't accidentally override `databaseLocked` and `proxyMetadataVersion`.

Reply:
done